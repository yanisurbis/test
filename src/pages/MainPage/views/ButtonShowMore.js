// component responsible for showing or hiding "Show More" button
// also for rendering loader when needed
import React, { PropTypes as t } from 'react'
import { Button } from 'semantic-ui-react'

const ButtonShowMore = ({ onClick, disabled }) => {
  return (
    <Button primary onClick={onClick} disabled={disabled}>
      Show More Issues
    </Button>
  )
}

ButtonShowMore.propTypes = {
  disabled: t.bool.isRequired,
  onClick: t.func.isRequired
}

export default ButtonShowMore
