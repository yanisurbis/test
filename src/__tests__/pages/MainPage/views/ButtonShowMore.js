/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ButtonShowMore from 'code/pages/MainPage/views/ButtonShowMore'

const createProps = disabled => ({
  disabled,
  onClick: jest.fn(),
});

test('<ButtonShowMore /> renders correctly when disabled is false', () => {
  const props = createProps(false);
  const wrapper = shallow(
    <ButtonShowMore {...props} />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

test('<ButtonShowMore /> renders correctly when disabled is true', () => {
  const props = createProps(true);
  const wrapper = shallow(
    <ButtonShowMore {...props} />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
