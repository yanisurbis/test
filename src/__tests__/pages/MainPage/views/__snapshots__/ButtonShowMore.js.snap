exports[`test <ButtonShowMore /> renders correctly when disabled is false 1`] = `
<Button
  as="button"
  disabled={false}
  onClick={[Function]}
  primary={true}>
  Show More Issues
</Button>
`;

exports[`test <ButtonShowMore /> renders correctly when disabled is true 1`] = `
<Button
  as="button"
  disabled={true}
  onClick={[Function]}
  primary={true}>
  Show More Issues
</Button>
`;
